apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,289 @@
+      schoolType: existingSchoolInfo.school_type || '',
+      schoolName: existingSchoolInfo.school_name || '',
+      schoolState: existingSchoolInfo.state || '',
+      schoolZip: existingSchoolInfo.zip || '',
+      schoolLocation: existingSchoolInfo.full_address || '',
+      ncesSchoolId: initialNcesSchoolId,
+      showSchoolInfoUnknownError: false,
+    };
+  }
+
+  logEvent(eventName, data = {}) {
+    firehoseClient.putRecord('analysis-events', {
+      study: 'school_info_interstitial',
+      study_group: 'control',
+      event: eventName,
+      data_json: JSON.stringify({
+        isComplete: SchoolInfoInterstitial.isSchoolInfoComplete(this.state),
+        ...data
+      }),
+    });
+  }
+
+  static isSchoolInfoComplete(state) {
+    if (!state.country) {
+      return false;
+    }
+
+    if (state.country !== 'United States') {
+      return true;
+    }
+
+    if (['homeschool', 'after school', 'organization', 'other'].includes(state.schoolType)) {
+      return true;
+    }
+
+    if (state.ncesSchoolId && state.ncesSchoolId !== '-1') {
+      return true;
+    }
+
+    return !!(
+      state.country
+      && state.schoolType
+      && state.schoolName
+      && (
+        (state.schoolState && state.schoolZip)
+        || state.schoolLocation
+      )
+    );
+  }
+
+  componentDidMount() {
+    this.logEvent(FIREHOSE_EVENTS.SHOW);
+  }
+
+
+  handleSchoolInfoSubmit = () => {
+    this.logEvent(FIREHOSE_EVENTS.SUBMIT, {
+      attempt: this.state.showSchoolInfoUnknownError ? 2 : 1
+    });
+
+    const isUS = this.state.country === 'United States';
+    let schoolData;
+    if (this.state.schoolType === '') {
+      schoolData = {};
+    } else if (isUS && SCHOOL_TYPES_HAVING_NCES_SEARCH.includes(this.state.schoolType)) {
+      if (this.state.ncesSchoolId === '-1') {
+        // I couldn't find my school.
+        schoolData = {
+          "user[school_info_attributes][school_name]": this.state.schoolName,
+          "user[school_info_attributes][school_state]": this.state.schoolState,
+          "user[school_info_attributes][school_zip]": this.state.schoolZip,
+          "user[school_info_attributes][full_address]": this.state.schoolLocation,
+        };
+      } else {
+        schoolData = {
+          "user[school_info_attributes][school_id]": this.state.ncesSchoolId,
+        };
+      }
+    } else {
+      schoolData = {
+        "user[school_info_attributes][school_name]": this.state.schoolName,
+        "user[school_info_attributes][full_address]": this.state.schoolLocation,
+      };
+    }
+
+    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
+    $.post({
+      url: formUrl + '.json',
+      dataType: "json",
+      data: {
+        '_method': 'patch',
+        [authTokenName]: authTokenValue,
+        "user[school_info_attributes][country]": this.state.country,
+        "user[school_info_attributes][school_type]": this.state.schoolType,
+        ...schoolData,
+      },
```

Comment:
If we have an NCES school id then we shouldn't be passing the country and school_type. That data lives on the school object and will be copied over by the code at https://github.com/code-dot-org/code-dot-org/blob/809add2fc858d184afe2b96844c84d0b32b12e9f/dashboard/app/models/school_info.rb#L82

Reply:
So if we have an NCES id, we're really _just_ sending that id and nothing else?

Reply:
Yes, just the school id. I'm not sure how the code evolved but right now a `school_info` is basically a generalization of a `school` that allows the data to exist without being tied to a `school`. In the case that there is a `school` then all of the data in `school _info` is derived from the `school`. It is unfortunate that some of the data is denormalized into the `school_info` itself in those cases. The `sync_from_schools` method was put in place to ensure that the denormalized data matched. We actually have a few places that are still sending in data that don't match the data associated with the provided `school_id`. Mostly, this tends to be `school_type` but is also sometimes `country` (`US` vs. `United States` again.)

Reply:
Got it.  I'll update this code and my test cases to make sure we are sending the minimum appropriate info, and letting `sync_from_schools` do its job.